### DISCUSSION TITLE
How do I implement statistical analysis with "Scope" as "InstanceEndpoint"

### DISCUSSION THREAD

**Discussion by cxntsh** [apache/skywalking#9829] [Q&A]
We all know that there are currently three dimensions of Scope statistics, namely Service, ServiceInstance, and Endpoint( ServiceEndpoint ), so when we need to see the statistics of a page under a certain version, how to implement streaming computing configuration through traffic source?

**Comment by wu-sheng** [comment]
Are you sure? Please run the math first. N service, M pods per service, Y endoints per service. Then you will have N x M x Y InstanceEndpoints. In a real env, this could be easily, 100 services x 50 pods/service * 1000 endpoints/service, you will face 5m InstanceEndpoints

**Reply by cxntsh** [reply]
Makes sense, but . . . . but . . . Thank you very much, I will think about how to convince the leader

**Reply by wu-sheng** [reply]
In the same case, you only have 5k instances and 10k endpoints. After your proposal, 5m. Think about which scale you could build a database to support 50 times larger dataset.

**Reply by cxntsh** [reply]
First of all, I really agree with your opinion, but if we put it into the field of web observation, we need to see the performance of different endpoints (urlPath) under different versions of the web build, but they still belong to the same service, I think many users may have similar needs.

**Reply by wu-sheng** [reply]
No matter what it is, this is project level global change.
According to above concern, this can't be applied. Monitoring is not for everything, it should cost reasonable resources to provide as much as possible.

**Reply by wu-sheng** [reply]
Besides data size, how to generate so long ID is another issue. 
The ElasticSearch ID is 512 bytes length. Endpoint ID is already encoded service name and endpoint name. If you put instance name in it, it would be too long.
So, like I said, this is a global issue, you can't change this easily.
